### PR TITLE
YM-402 | Fix mobile navigation by adopting navigation component from HDS

### PR DESCRIPTION
**Pull request by Jokinen**
## Description

This change replaces our own navigation implementation with that of HDS. By doing this, we are able to tackle a few accessibility issues.
* The menu mobile menu becomes usable with screen readers
* A skip to content link will be added
* The problematic labelling of the logo and service name will be fixed

## Context

[YM-402](https://helsinkisolutionoffice.atlassian.net/browse/YM-402)

## How Has This Been Tested?

I've ran the test suite against these changes and checked that it passes. I've manually tested the new menu.

## Manual Testing Instructions for Reviewers

With the mobile layout and screen reader

1. Expect the mobile menu to have an accessible name
2. Expect the mobile menu to have aria roles which denote whether it is open or not
3. Expect the mobile menu content to be accessible

With a screen reader

1. Expect the logo image and application title to have a single shared label ("Helsinki: Jässäri")

With a screen reader

1. Expect the first focusable element to be a jump to content button
2. Expect the button to skip the menu and jump to content

With a screen reader

1. Expect language items to have correct lang attributes

With a browser

1. Start creating a new profile
2. Expect to see a sign out button
3. Fill in details
4. Open approval mail
5. Expect user menu to be hidden
6. Expect logo and app title to lead to the same page

With a browser

1. Sign in
2. Expect user menu to contain link to Helsinki-profile and logging out

### CODE REVIEW

File: src/common/components/layout/header/Header.module.css [hunk 1]
```diff
@@ -1,57 +1,16 @@
...
   grid-row: navigation;
-  padding: 10px 0;
-}
-.header > * {
-  grid-column: container;
 }
 
-.content {
-  align-items: center;
+.fakeAction {
```
Comment (Jokinen) [inline src/common/components/layout/header/Header.module.css:6]: By default HDS does not support buttons in the actions section of the menu. Note to self: test this on mobile after profile deletion works again.

File: src/i18n/en.json [hunk 2]
```diff
@@ -108,7 +108,10 @@
     "menuButtonLabel": "Profile menu",
     "profile": "Helsinki Profile",
     "signin": "Log in",
-    "signout": "Log out"
+    "signout": "Log out",
+    "menu": "Menu",
+    "skipToConten": "Skip to main content",
```
Comment (tuovinensanttu) [inline src/i18n/en.json:113]: `skipToContent` has a typo in it, this causes the translation not to work on the button.

File: src/common/components/layout/PageContent.tsx [hunk 3]
```diff
@@ -5,6 +5,8 @@ import { useMatomo } from '@datapunt/matomo-tracker-react';
 import LoadingContent from '../loading/LoadingContent';
 import styles from './pageContent.module.css';
 
+export const MAIN_CONTENT_ID = 'main-content';
+
```
Comment (tuovinensanttu) [inline src/common/components/layout/PageContent.tsx:9]: This solution triggers a navigation to `/main-content` and the user sees `404 - page`.

File: src/common/components/layout/header/Header.module.css (hunk 1 above)
Comment (Jokinen) [reply src/common/components/layout/header/Header.module.css:6]: This seems to be working fine on mobile

**Comment by Jokinen** [comment]
I completed the last bit of manual testing for this change now and everything looked to be working as expected.

**Review by tuovinensanttu — APPROVED** [review]
Everything seems to be working 👌

**Pull request merged**